src/content/en/updates/posts/2016/05/houdini.md

```diff
@@ -0,0 +1,366 @@
+            const y = Math.random()*height;
+            const constraintSubSpace = new ConstraintSpace();
+            constraintSubSpace.width = width-x;
+            constraintSubSpace.height = height-y;
+            const childFragment = child.doLayout(constraintSubSpace);
+            childFragment.x = x;
+            childFragment.y = y;
+        }
+
+        return {
+            minContent: 0,
+            maxContent: 0,
+            width: width,
+            height: height,
+            fragments: [],
+            unPositionedChildren: [],
+            breakToken: null
+        };
+    }
+});
+{% endhighlight %}
+
+### Typed CSSOM ([spec][Typed CSSOM spec])
+That’s enough worklets for now, let’s get back to some good ol’ APIs. Typed
+CSSOM (CSS Object Model (Cascading Style Sheets Object Model)) addresses a
+problem we probably all have encountered and just learned to just put up with.
+Let me illustrate with a line of JavaScript:
+
+{% highlight JS %}
+$('#someDiv').style.height = getRandomInt() + 'px';
+{% endhighlight %}
+
+We are doing math, converting a number to a string to append a unit just to have
+the browser parse that string and convert it back to a number for the CSS engine
+to use. This gets even uglier when you [manipulate transforms with
+JavaScript][Aerotwist FLIP]. No more! CSS is about to get some typing! This
+draft is one of the more mature ones and a [polyfill][Typed CSSOM polyfill] is
+actually already being worked on (Disclaimer: Using the polyfill will obviously
+add *even more* computational overhead. The point is to show how convenient the
+API is).
+
+Instead of strings you will be working on an element’s `StylePropertyMap`, where
+each CSS attribute has it’s own key and corresponding value type. Attributes
+like `width` have `LengthValue` as their value type. A `LengthValue` is a
+dictionary of all CSS units like `em`, `rem`, `px`, `percent`, etc. Setting
+`height: calc(5px + 5%)` would yield a `LengthValue{px: 5, percent: 5}`. Some
+properties like `box-sizing` just accept certain keywords and therefore have a
+`KeywordValue`. The validity of those attributes could now be checked at
+runtime.
+
+{% highlight JS %}
+var w1 = $('#div1').styleMap.get('width');
+var w2 = $('#div2').styleMap.get('width');
+$('#div3').styleMap.set('background-size',
+  [new SimpleLength(200, 'px'), w1.add(w2)])
+$('#div4')).styleMap.get('margin-left')
+  // => {em: 5, percent: 50}
+{% endhighlight %}
+
+### Font Metrics
+
+Font metrics is exactly what it sounds like. What is the bounding box (or the
+bounding boxes when we are wrapping) when I render string X with font Y at size
+Z? What if I go all crazy unicode on you like using [ruby annotations]? This has
+been requested a lot in the past and Houdini should finally make these wishes
+come true.
+
+### Properties and Values ([spec][Properties and Values spec])
+
+Do you know CSS Custom Properties (or their unofficial alias “CSS Variables”)?
```

Comment:
Consider linking to https://developers.google.com/web/updates/2016/02/css-variables-why-should-you-care